src/components/VmDetails/cards/DetailsCard/index.js

```diff
@@ -251,35 +257,37 @@ class DetailsCard extends React.Component {
       let fieldUpdated
       switch (fieldName) {
         case 'cluster':
-          updates = updates.set('cluster', this.props.clusters.get(value))
+          updates = updates.set('cluster', clusters.get(value))
           fieldUpdated = 'cluster'
 
           // Change the template to 'Blank' if the VM's template isn't in the new cluster
           {
-            const template = this.props.templates.get(updates.getIn(['template', 'id']))
+            const template = templates.get(updates.getIn(['template', 'id']))
             if (template && template.get('clusterId') && template.get('clusterId') !== value) {
-              changeQueue.push({ fieldName: 'template', value: this.props.blankTemplateId })
+              changeQueue.push({ fieldName: 'template', value: blankTemplateId })
             }
           }
           break
 
         case 'template':
-          updates = updates.set('template', this.props.templates.get(value))
+          updates = updates.set('template', templates.get(value))
           fieldUpdated = 'template'
 
           // Apply settings from the template to the VM (memory, CPUs, OS, cloudInit, bootMenuEnabled)
           {
             const template = this.props.templates.get(value)
             if (template) {
               const templateOsType = template.getIn(['os', 'type'], 'other')
-              const templateOs = this.props.operatingSystems.find(os => os.get('type') === templateOsType)
+              const oldOS = operatingSystems.find(os => os.get('name') === templateOsType)
```

Comment:
IMHO this line contains the fix - `os.name` and `template.os.type` seem to be OS unique name.
Check both REST definitions:
1.  `os.name` is set [here](https://github.com/oVirt/ovirt-engine/blob/2314e3b31e42ff1c96395948af7a9a7f996f56ab/backend/manager/modules/restapi/jaxrs/src/main/java/org/ovirt/engine/api/restapi/resource/BackendOperatingSystemResource.java#L36)
2. `template.os.type` is set [here](https://github.com/oVirt/ovirt-engine/blob/2314e3b31e42ff1c96395948af7a9a7f996f56ab/backend/manager/modules/restapi/types/src/main/java/org/ovirt/engine/api/restapi/types/TemplateMapper.java#L90)

Why not just:
```
const templateOs = operatingSystems.find(os => os.get('name') === templateOsType) || findDefaultForCluster()
```
?
Of course `findDefaultForCluster()` remains the tricky part :)